common/debug-interface/src/libra_trace.rs

```diff
@@ -296,3 +342,58 @@ fn abbreviate_crate(name: &str) -> &str {
         _ => name,
     }
 }
+
+// This is exact copy of similar function in log crate
+/// Sets libra trace config
+pub fn set_libra_trace(config: &HashMap<String, String>) -> Result<()> {
+    match parse_sampling_config(config) {
+        Ok(sampling) => unsafe {
+            match LIBRA_TRACE_STATE.compare_and_swap(UNINITIALIZED, INITIALIZING, Ordering::SeqCst)
+            {
+                UNINITIALIZED => {
+                    SAMPLING_CONFIG = Some(sampling);
+                    LIBRA_TRACE_STATE.store(INITIALIZED, Ordering::SeqCst);
+                    Ok(())
+                }
+                INITIALIZING => {
+                    while LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZING {}
+                    bail!("Failed to initialize LIBRA_TRACE_STATE");
+                }
+                _ => bail!("Failed to initialize LIBRA_TRACE_STATE"),
+            }
+        },
+        Err(s) => bail!("Failed to parse sampling config: {}", s),
+    }
+}
+
+fn parse_sampling_config(config: &HashMap<String, String>) -> Result<Sampling> {
+    let mut map = HashMap::new();
+    for (k, v) in config {
+        let k: &'static str = Box::leak(k.clone().into_boxed_str());
+        let v = v.split('/').collect::<Vec<&str>>();
+        let v = CategorySampling {
+            nominator: v[0].parse::<u64>()?,
+            denominator: v[1].parse::<u64>()?,
+        };
+        map.insert(k, v);
+    }
+    Ok(Sampling(map))
+}
+
+/// Checks if libra trace is enabled
+pub fn libra_trace_set() -> bool {
+    LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZED
+}
+
+pub fn is_selected(node: (&'static str, u64)) -> bool {
```

Comment:
Can we check `libra_trace_set` here?
Currently you do it in every call site before is_selected, there are few problems with that
a) Code duplication
b) In the future, someone might forgot to check libra_trace_set before calling is_select, which is unsafe.